lib/v2/datasource/remote/api/signup_repository.dart

```diff
@@ -52,4 +52,35 @@ class SignupRepository extends NetworkRepository {
       }
     }
   }
+
+  /// Checks the availability of the given username on the blockchain, it will return the success response if the username
+  /// is taken, otherwise returns an error
+  Future<Result> isUsernameTaken(String username) async {
+    final keyAccountsURL = '$baseURL/v1/chain/get_account';
+
+    final requestBody = '{ "account_name": "$username" }';
+
+    return await http
+        .post(
+          Uri.parse(keyAccountsURL),
+          body: requestBody,
+        )
+        .then((http.Response response) => mapHttpResponse(response, (dynamic body) {
+              // Username exists on the blockchain and is not available
+              return null;
+            }))
+        .catchError((error) => mapHttpError(error));
+  }
+
+  String generateUsername(String fullname) {
```

Comment:
Should this method be in the repo? I dont see any network calls on this method.

Reply:
Moved to state mapper.